server/polar/customer_portal/endpoints/subscription.py

```diff
@@ -104,6 +105,72 @@ async def get(
     return subscription
 
 
+class SubscriptionChargePreviewResponse(TypedDict):
+    base_amount: int
+    subtotal_amount: int
+    tax_amount: int
+    total_amount: int
+
+
+@router.get(
+    "/{id}/charge-preview",
+    summary="Preview Next Charge For Active Subscription",
+    response_model=SubscriptionChargePreviewResponse,
+    responses={404: SubscriptionNotFound},
+)
+async def get_charge_preview(
+    id: SubscriptionID,
+    auth_subject: auth.CustomerPortalRead,
+    session: AsyncSession = Depends(get_db_session),
+) -> SubscriptionChargePreviewResponse:
+    """Get current period usage and cost breakdown for a subscription."""
+    subscription = await customer_subscription_service.get_by_id(
+        session, auth_subject, id
+    )
+
+    if subscription is None:
+        raise ResourceNotFound()
+
+    if subscription.status != "active":
+        ## FIXME Is a 404 the correct behavior?
+        raise ResourceNotFound()
+
+    base_price = next(p for p in subscription.prices if p.amount_type == "fixed")
```

Comment:
This won't work with Custom (PWYW) and Free prices. I suggest you go through `subscription_product_prices` and use amount instead:

```py
base_price = sum(p.amount for p in subscription.subscription_product_prices)
```

`subscription.prices` is actually a proxy shortcut that "hop" through `subscription_product_prices`. But we do store the "real" amount directly in `subscription_product_prices`.